src/swupd/manifest_test.go

```diff
@@ -257,3 +262,117 @@ func TestInvalidManifests(t *testing.T) {
 		})
 	}
 }
+
+func compareFiles(file1path string, file2path string) (bool, error) {
+	var err error
+	var f1Stat, f2Stat os.FileInfo
+	var f1, f2 *os.File
+
+	chunkSize := 65536
+
+	f1Stat, err = os.Lstat(file1path)
+	if err != nil {
+		return false, err
+	}
+
+	f2Stat, err = os.Lstat(file2path)
+	if err != nil {
+		return false, err
+	}
+
+	if f1Stat.Size() != f2Stat.Size() {
+		return false, nil
+	}
+
+	f1, err = os.Open(file1path)
+	if err != nil {
+		return false, err
+	}
+	defer f1.Close()
+
+	f2, err = os.Open(file2path)
+	if err != nil {
+		return false, err
+	}
+	defer f2.Close()
+
+	b1 := bufio.NewReader(f1)
+	b2 := bufio.NewReader(f2)
+	for {
+		bytesRead1 := make([]byte, chunkSize)
+		b1BytesIn, err1 := b1.Read(bytesRead1)
+
+		bytesRead2 := make([]byte, chunkSize)
+		b2BytesIn, err2 := b2.Read(bytesRead2)
+
+		if b1BytesIn != b2BytesIn {
+			return false, nil
+		}
+
+		if err1 != nil || err2 != nil {
+			if err1 == io.EOF && err2 == io.EOF {
+				return true, nil
+			} else if err1 == io.EOF || err2 == io.EOF {
+				return false, nil
+			} else {
+				return false, fmt.Errorf("%v - %v", err1, err2)
+			}
+		}
+
+		if !bytes.Equal(bytesRead1, bytesRead2) {
+			return false, nil
+		}
+	}
+}
+
+func TestWriteManifestFile(t *testing.T) {
+	path := "testdata/manifest.good"
+
+	var m Manifest
+	if err := m.ReadManifestFromFile(path); err != nil {
+		t.Fatal(err)
+	}
+
+	if len(m.Files) == 0 {
+		t.Fatal("ReadManifestFromFile did not add file entried to the file list")
+	}
+
+	newpath := "testdata/manifest.result"
+	if err := m.WriteManifestFile(newpath); err != nil {
+		t.Error(err)
+	}
+
+	match, err := compareFiles(path, newpath)
+	if err != nil {
+		t.Fatal("unable to compare old and new manifest")
+	}
+
+	if !match {
+		t.Errorf("generated %v did not match read %v file", newpath, path)
+	}
+
+	err = os.Remove(newpath)
+	if err != nil {
+		t.Error("unable to remove file, did it not close properly?")
+	}
+}
+
+func TestGetManifestTimeStamp(t *testing.T) {
+	now := time.Now()
+	m := Manifest{Header: ManifestHeader{TimeStamp: now}}
+	if m.getManifestTimeStamp() != now.Unix() {
+		t.Error("gerManifestTimestamp returned unexpected result")
+	}
+}
+
+func TestWriteManifestFileBadHeader(t *testing.T) {
+	m := Manifest{Header: ManifestHeader{}}
+	path := "testdata/manifest.result"
```

Comment:
Minor: use ioutil.TempFile to create a temporary file (you can grab the Name() from the returned file). That way we don't mix testdata input with temporary stuff and we don't need to care about tests using different temporary names.